### DISCUSSION TITLE
Confused about choice of Redis vs ActiveRecord

### DISCUSSION THREAD

**Discussion by Bandes** [jnunemaker/flipper#658] [Q&A] [answered]
I want to make sure I'm making an informed choice when choosing a storage mechanism for Flipper. I have Redis and Postgres in my application stack, so both Redis and ActiveRecord are options for me. What are the pros/cons of these options? My guess is that Redis is more performant but doesn't guarantee persistence as well? Is there a reason for choosing one over the other for a product in production?

**Comment by jnunemaker** [comment, marked as answer]
It's really just personal preference or project preference. Maybe a project doesn't have redis but has active record or vice versa. There isn't really a performance difference unless your setup has a beefier redis or postgres instance or faster networking to one of those. 

If you have both, I'd go with active record/postgres. Someday if that isn't performant enough you can always put active support (or memcache or redis) cache in front.
